### ISSUE TITLE
SIFT no longer available in opencv > 3.4.2

### ISSUE THREAD

**Issue by bouweandela** [DynaSlum/satsense#13]
From [travis](https://travis-ci.com/DynaSlum/satsense/jobs/151158595):

cv2.error: OpenCV(3.4.3) /io/opencv_contrib/modules/xfeatures2d/src/sift.cpp:1207: error: (-213:The function/feature is not implemented) This algorithm is patented and is excluded in this configuration; Set OPENCV_ENABLE_NONFREE CMake option and rebuild the library in function 'create'

**Comment by bouweandela** [comment]
Related https://github.com/skvark/opencv-python/issues/126

**Comment by bouweandela** [comment]
Our current solution is to pin opencv to versions smaller than 3.4.3, because those supported SIFT. Let's see how this develops upstream. We may have to drop support for SIFT or get it from another library in the future.

**Comment by yonelay11** [comment]
I also faced this issues. 

> pip install opencv-python==3.4.2.16

> pip install opencv-contrib-python==3.4.2.16
this opencv_3.4.2 version handles this issues. This is good for me.

**Comment by hm43** [comment]
> I also faced this issues.
> 
> > pip install opencv-python==3.4.2.16
> 
> > pip install opencv-contrib-python==3.4.2.16
> > this opencv_3.4.2 version handles this issues. This is good for me.

When try to install that version I get:
```
ERROR: Could not find a version that satisfies the requirement opencv-python==3.4.2.16 (from versions: 3.4.8.29, 4.1.2.30)                                                                                                                        
ERROR: No matching distribution found for opencv-python==3.4.2.16
```

**Comment by theRealSuperMario** [comment]
Is there any reasonable explanation on why they kicked out sift?

**Comment by lima36** [comment]
opencv 3.4.2.17 is available but there is error in the 4.2.0
